resources/indexer.py

```diff
@@ -314,20 +312,209 @@ def _get_i3_part_file_metadata(file, site, run_meta_xml=None, gaps_dict=None, gc
+            self.subrun = 0
+            p = self.file.name.split('Part')[1]
+            self.part = int(p.split('.')[0])
+
+        # Ex: Level2a_IC59_data_Run00115968_Part00000290.i3.gz
+        elif re.match(r'(.*)(_IC.*)_data_Run[0-9]+_Part[0-9]+(.*)', self.file.name):
+            n = self.file.name.split('IC')[1]
+            strings = n.split('_')[0]
+            self.season_year = IceCubeSeason.name_to_year(f"IC{strings}")
+            self.subrun = 0
+            p = self.file.name.split('Part')[1]
+            self.part = int(p.split('.')[0])
+
+        # Ex: Level2_All_Run00111562_Part00000046.i3.gz
+        elif re.match(r'(.*)_Run[0-9]+(.*)_Part[0-9]+(.*)', self.file.name):
+            self.season_year = None
+            self.subrun = 0
+            p = self.file.name.split('Part')[1]
+            self.part = int(p.split('.')[0])
+
+        else:
+            raise Exception(f"Filename not in a known L2 file format, {self.file.name}.")
+
+    @staticmethod
+    def is_file(file):
+        """True if the file is in the [...#].i3[...] file format."""
+        # Ex: Level2_IC86.2017_data_Run00130484_Subrun00000000_00000188.i3.zst
+        # check if last char of filename (w/o extension) is an int
+        return (".i3" in file.name) and (file.name.split('.i3')[0][-1]).isdigit()
+
+
+class PFFiltFileMetadata(I3FileMetadata):
+    """Metadata for PFFilt i3 files"""
+
+    def __init__(self, file, site):
+        super().__init__(file, site, ProcessingLevel.PFFilt)
+        with tarfile.open(file.path) as tar:
+            for tar_obj in tar:
+                if ".meta.xml" in tar_obj.name:
+                    self.meta_xml = xmltodict.parse(tar.extractfile(tar_obj))
+
+    def generate(self):
+        """Gather the file's metadata."""
+        metadata = super().generate()
+        return metadata
+
+    def _parse_filepath(self):
+        """Set the year, run, subrun, and part from the file name."""
+        self.run = I3FileMetadata.parse_run_number(self.file)
+
+        # Ex: PFFilt_PhysicsFiltering_Run00131989_Subrun00000000_00000295.tar.bz2
+        # Ex: PFFilt_PhysicsTrig_PhysicsFiltering_Run00121503_Subrun00000000_00000314.tar.bz2
+        if re.match(r'PFFilt_(.*)_Run[0-9]+_Subrun[0-9]+_[0-9]+(.*)', self.file.name):
+            self.season_year = None
+            s = self.file.name.split('Subrun')[1]
+            self.subrun = int(s.split('_')[0])
+            p = self.file.name.split('_')[-1]
+            self.part = int(p.split('.')[0])
+        else:
+            raise Exception(f"Filename not in a known PFFilt file format, {self.file.name}.")
+
+    @staticmethod
+    def is_file(file):
+        """ True if the file is in the [...#].tar.bz2 file format. """
+        # Ex. PFFilt_PhysicsFiltering_Run00131989_Subrun00000000_00000295.tar.bz2
+        # check if last char of filename (w/o extension) is an int
+        return (".tar.bz2" in file.name) and (file.name.split('.tar.bz2')[0][-1]).isdigit()
```

Comment:
I'm pretty sure PFFilt files changed from `.tar.gz` to `.tar.bz2` at some point, so this may return the wrong thing if you're working with old/ancient PFFilt files.

Reply:
This is a good point.  Assume the compression can change, to include '.zst' or any other format.

Reply:
Good catch. In 2006 they're `.tar.gz`. I'll revise this.

Reply:
I'm adding support for `.tar.gz`, `.tar.bz2`, `.tar.zst`.